core/cloudflow-akka/src/main/scala/cloudflow/akkastream/persistence/TableCreator.scala

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.akkastream.persistence
+
+// Adopted from io.cloudstate.proxy.jdbc.SlickEnsureTablesExistReadyCheck, which is adopted from
+// from https://github.com/lagom/lagom/blob/60897ef752ddbfc28553d3726b8fdb830a3ebdc4/persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickProvider.scala
+
+import java.sql.Connection
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.persistence.jdbc.config.{ ConfigKeys, _ }
+import akka.persistence.jdbc.journal.dao.JournalTables
+import akka.persistence.jdbc.snapshot.dao.SnapshotTables
+import akka.persistence.jdbc.util.SlickExtension
+import com.typesafe.config._
+import net.ceedubs.ficus.Ficus._
+import org.slf4j.LoggerFactory
+import slick.jdbc._
+import slick.jdbc.meta.MTable
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+import scala.util._
+
+object TableCreator {
+
+  private val SLICK_DATABASE  = "akka-persistence-jdbc.shared-databases.slick"
+  private val AUTO_CREATE_KEY = "auto-create-tables"
+  private val log             = LoggerFactory.getLogger(this.getClass)
+
+  def createTables(system: ActorSystem): Unit = {
+
+    val jdbc = system.settings.config.as[Option[Config]](SLICK_DATABASE)
+    jdbc match {
+      case Some(config) => // This is JDBC
+        val create = config.as[Option[Boolean]](AUTO_CREATE_KEY).getOrElse(false)
+        if (create)
+          createTablesInternal(system)
+      case _ =>
+    }
+  }
+
+  private def createTablesInternal(system: ActorSystem): Unit = {
```

Comment:
I recommend against the auto-creation of database tables.

From APJDBC 5.0.0-RC1 it offers a test kit API to create tables.

https://github.com/akka/akka-persistence-jdbc/blob/master/core/src/main/scala/akka/persistence/jdbc/testkit/scaladsl/SchemaUtils.scala

The **new** schemas are included as resources in the jar.